contrib/runners/orquesta_runner/orquesta_runner/orquesta_runner.py

```diff
@@ -82,7 +83,22 @@ def _construct_st2_context(self):
 
         return st2_ctx
 
+    def _get_rerun_options(self):
+        return self.context.get('re-run', {})
+
     def run(self, action_parameters):
+        rerun_options = self._get_rerun_options()
+        tasks = rerun_options.get('tasks', None)
+        rerun = self.rerun_ex_ref and tasks
+
+        if rerun:
+            result = self.rerun_workflow(ex_ref=self.rerun_ex_ref, rerun_options=tasks)
+        else:
+            result = self.start_workflow(action_parameters=action_parameters)
+
+        return result
```

Comment:
Why can't we share the following code snippet from the original `run` method here?

```
        except wf_exc.WorkflowInspectionError as e:
            status = ac_const.LIVEACTION_STATUS_FAILED
            result = {'errors': e.args[1], 'output': None}
            return (status, result, self.context)
        except Exception as e:
            status = ac_const.LIVEACTION_STATUS_FAILED
            result = {'errors': [{'message': six.text_type(e)}], 'output': None}
            return (status, result, self.context)
        if wf_ex_db.status in wf_statuses.COMPLETED_STATUSES:
            status = wf_ex_db.status
            result = {'output': wf_ex_db.output or None}
            if wf_ex_db.status in wf_statuses.ABENDED_STATUSES:
                result['errors'] = wf_ex_db.errors
            for wf_ex_error in wf_ex_db.errors:
                msg = '[%s] Workflow execution completed with errors.'
                LOG.error(msg, str(self.execution.id), extra=wf_ex_error)
            return (status, result, self.context)
        # Set return values.
        status = ac_const.LIVEACTION_STATUS_RUNNING
        partial_results = {}
        ctx = self._construct_context(wf_ex_db)
	
	return (status, partial_results, ctx)
```